### DISCUSSION TITLE
Locally-installed components not availabe in mobile application

### DISCUSSION THREAD

**Discussion by julianfairfax** [standardnotes/forum#2889] [General]
**Describe the bug**
When you disable the use hosted option for an extension (editor in this case) in the mobile application, it shows an error saying that locally-installed components are not available in the web application.

**To Reproduce**
1. Install an extension
2. Disable the use hosted option for this extension
3. Try using it (if it's an editor)

**Expected behavior**
The locally-installed extension should work. As this is obviously not the web application.

**Screenshots**
![Screenshot_20221103-184040](https://user-images.githubusercontent.com/43753131/199795540-cea4e0be-a191-4826-87c5-8aef739e6bdf.png)

**Smartphone (please complete the following information):**
 - Device: Google Pixel 3a
 - OS: Android 12 (GrapheneOS)
 - Version 3.45.5

**Comment by julianfairfax** [comment]
@effieeee could you or someone else look into this at some point?

**Comment by effieeee** [comment]
Hi @julianfairfax, it looks like 3rd party extensions are also cached on the mobile apps now, so the app should be able to load either the hosted or local copy, regardless of the toggle in the Advanced settings

**Comment by julianfairfax** [comment]
> Hi @julianfairfax, it looks like 3rd party extensions are also cached on the mobile apps now, so the app should be able to load either the hosted or local copy, regardless of the toggle in the Advanced settings

Even in the latest mobile release from GitHub, this doesn't seem to be the case. I can tell, because when the extension is loaded from the internet, it's loaded after themes, and so no third-party editor will work with the built-in dark theme.

**Comment by effieeee** [comment]
Hmm, I did also try using the dark theme while applying a 3rd party editor (Free Draw and Rich Markdown Editor) on mobile, and although Free Draw understandably doesn't adapt to the dark theme, both of the editors were still able to load the cached copy while offline.

I'd like to clarify as well that it's ok to leave the toggle enabled in the advanced settings, because once you've verified that your 3rd party extension works offline, you're assured that the local installation is always available for use.
